UPGRADE.rst

```diff
@@ -55,50 +55,140 @@ Upgrading to v1.4.0
 Config options
 --------------
 
-**Note: Registration by email address or phone number will not work in this release unless
-some config options are changed from their defaults.**
-
-This is due to Synapse v1.4.0 now defaulting to sending registration and password reset tokens
-itself. This is for security reasons as well as putting less reliance on identity servers.
-However, currently Synapse only supports sending emails, and does not have support for
-phone-based password reset or account registration. If Synapse is configured to handle these on
-its own, phone-based password resets and registration will be disabled. For Synapse to send
-emails, the ``email`` block of the config must be filled out. If not, then password resets and
-registration via email will be disabled entirely.
-
-This release also deprecates the ``email.trust_identity_server_for_password_resets`` option and
-replaces it with the ``account_threepid_delegates`` dictionary. This option defines whether the
-homeserver should delegate an external server (typically an `identity server
-<https://matrix.org/docs/spec/identity_service/r0.2.1>`_) to handle sending password reset or
-registration messages via email and SMS.
+**Note: Registration by email address or phone number will not work in this
+release unless some config options are changed from their defaults.**
+
+Previous versions of Synapse delegate the sending of registration emails and
+SMS to an identity server by default. In most cases this server is vector.im or
+matrix.org.
+
+In Synapse 1.4, for security and privacy reasons, the homeserver will no longer
+delegate email or SMS to an identity server by default and instead the admin
+will need to explicitly decide how they would like email and SMS to be sent.
+
+In the medium term the vector.im and matrix.org identity servers will disable
+sending email and SMS entirely, however in order to ease the transition they
+will retain the capability to send email and SMS for a limited period. Email
+will be disabled on XXX (giving 2 months notice), disabling SMS will follow
+some time after that once SMS sending support lands in Synapse.
+
+Once email and SMS support has been disabled in the vector.im and matrix.org
+identity servers, all Synapse versions that depend on those instances will be
+unable to send email and SMS through them. There are no imminent plans to
+remove email and SMS support from Sydent altogether.
+
+Email
+-----
+
+Following upgrade, to continue using email as a registration method admins can
+either:-
+
+* Configure Synapse to use an alternate email server (details follow).
+* Run or choose an identity server (with email support) and delegate to it.
```

Comment:
all ISes have "email support". @anoadragon453 what exactly is it that the IS has to support?

Reply:
Sending an email with a link featuring a validation token, client_secret and sid that when clicked hits an endpoint on the identity server with those details, thus confirming validation.

Basically it needs to support the [/validate/email/requestToken](https://matrix.org/docs/spec/identity_service/unstable#post-matrix-identity-v2-validate-email-requesttoken) Identity Service API endpoint.

Reply:
The point is that some ISs will drop this support for delegation over time